pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java

```diff
@@ -1849,4 +1850,62 @@ public void testReadCommittedWithReadCompacted() throws Exception{
         Assert.assertEquals(messages, List.of("V2", "V3"));
     }
 
+
+    @Test
+    public void testReadCommittedWithCompaction() throws Exception{
```

Comment:
Should we add a test to send a replicated subscription snapshot marker and test the compaction works?

Reply:
I add test `testReplicatedSubscriptionWithCompaction`, PLAT~